plugins/modules/configuration_item.py

```diff
@@ -321,42 +325,62 @@ def build_payload(module, table_client):
 
 def ensure_present(module, table_client, attachment_client):
     mapper = get_mapper(module, "configuration_item_mapping", PAYLOAD_FIELDS_MAPPING)
-    query = utils.filter_dict(module.params, "sys_id")
+    query_sys_id = utils.filter_dict(module.params, "sys_id")
+    query_name = utils.filter_dict(module.params, "name")
     payload = build_payload(module, table_client)
     attachments = attachment.transform_metadata_list(
         module.params["attachments"], module.sha256
     )
 
-    if not query:
-        cmdb_table = module.params["sys_class_name"] or "cmdb_ci"
-
-        if not module.params["name"]:
-            raise errors.ServiceNowError("Missing required parameter: name")
+    if not query_sys_id:
+        configuration_item = table_client.get_record("cmdb_ci", query_name)
         # User did not specify existing CI, so we need to create a new one.
-        new = mapper.to_ansible(
-            table_client.create_record(
-                cmdb_table, mapper.to_snow(payload), module.check_mode
+        if not configuration_item:
+            cmdb_table = module.params["sys_class_name"] or "cmdb_ci"
+            new = mapper.to_ansible(
+                table_client.create_record(
+                    cmdb_table, mapper.to_snow(payload), module.check_mode
+                )
             )
-        )
-
-        # When we execute in check mode, new["sys_id"] is not defined.
-        # In order to give users back as much info as possible, we fake the sys_id in the
-        # next call.
-        new["attachments"] = attachment_client.upload_records(
-            cmdb_table,
-            new.get("sys_id", "N/A"),
-            attachments,
-            module.check_mode,
-        )
+            # When we execute in check mode, new["sys_id"] is not defined.
+            # In order to give users back as much info as possible, we fake the sys_id in the
+            # next call.
+            new["attachments"] = attachment_client.upload_records(
+                cmdb_table,
+                new.get("sys_id", "N/A"),
+                attachments,
+                module.check_mode,
+            )
+            return True, new, dict(before=None, after=new)
 
-        return True, new, dict(before=None, after=new)
+        else:
+            # Get existing record using provided name
+            old = mapper.to_ansible(table_client.get_record("cmdb_ci", query_name))
```

Comment:
Can you please create a variable for `cmdb_ci` so that it will be easily manageable?